homeassistant/components/mysensors/light.py

```diff
@@ -148,7 +147,7 @@ def _turn_on_rgb_and_w(self, hex_template: str, **kwargs: Any) -> None:
             return
         if new_rgb is not None:
             rgb = list(new_rgb)
-        if hex_template == "%02x%02x%02x%02x":
+        if hex_template == "%02x%02x%02x%02x" and white is not None:
```

Comment:
What happens if we get here with `new_white` and `white` is `None`?
Maybe it is better to change here:
https://github.com/home-assistant/core/blob/1afcad7ffbd7bfbdec9d5e935c0b88ae1982d422/homeassistant/components/mysensors/light.py#L153
`elif white is not None`
?